python_modules/dagster/dagster/_core/definitions/decorators/asset_check_decorator.py

```diff
@@ -232,19 +229,30 @@ def inner(fn: AssetCheckFunction) -> AssetChecksDefinition:
             retry_policy=retry_policy,
         )(fn)
 
-        checks_def = AssetChecksDefinition(
+        return AssetChecksDefinition(
+            keys_by_input_name={
+                input_tuple[0]: asset_key
+                for asset_key, input_tuple in input_tuples_by_asset_key.items()
+            },
+            keys_by_output_name={},
             node_def=op_def,
+            partitions_def=None,
+            partition_mappings=None,
+            asset_deps=None,
+            selected_asset_keys=None,
+            can_subset=False,
             resource_defs=wrap_resources_for_execution(resource_defs),
-            specs=[spec],
-            input_output_props=AssetChecksDefinitionInputOutputProps(
-                asset_keys_by_input_name={
-                    input_tuple[0]: asset_key
-                    for asset_key, input_tuple in input_tuples_by_asset_key.items()
-                },
-                asset_check_keys_by_output_name={op_def.output_defs[0].name: spec.key},
-            ),
+            group_names_by_key=None,
+            metadata_by_key=None,
+            tags_by_key=None,
+            freshness_policies_by_key=None,
+            auto_materialize_policies_by_key=None,
+            backfill_policy=None,
+            descriptions_by_key=None,
+            check_specs_by_output_name={op_def.output_defs[0].name: spec},
+            selected_asset_check_keys=None,
+            is_subset=False,
+            owners_by_key=None,
```

Comment:
nit: I'd recommend that the subclass fill in these defaults or that there is a function somewhere that does it. We want to support the use case for people programmatically generating these.

Reply:
I experimented with this but it creates a hassle around `with_resources` and copying. Also, I think we should not currently direct people building these in factories to use `AssetChecksDefinition`, since this might go away (needs discussion-- IMO it should)-- we should just tell them to use `@asset_check`.